app/javascript/miq-redux/reducer.ts

```diff
@@ -0,0 +1,25 @@
+import {createStore, Store, Reducer, Action} from 'redux';
+
+import { IMiqAction, AppReducer, IMiqReducerHash, AppState} from '../packs/miq-redux';
+
+const reducers: Set<AppReducer> = new Set();
+
+export const rootReducer: AppReducer = (state = {}, action: IMiqAction) => {
+    let newState = state;
+
+    reducers.forEach((reducer) => {
+        newState = reducer(newState, action)
```

Comment:
Thanks for the detailed answer. I don't see the two conflict. E.g. we should have an API to add redcuers (or a whole bunch of reducers)   but then pass it (array of reducer functions) as argument to combine reducers which will do the actual looping and basic tests (return values etc).

This will easily allow you to create sub reducers naturally (which I can't are us going without)